dnf-repoconfig-daemon/repoconf.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "configuration.hpp"
+#include "repoconf.hpp"
+
+#include <iostream>
+#include <sdbus-c++/sdbus-c++.h>
+#include <string>
+
+
+RepoConf::RepoConf(sdbus::IConnection &connection) : connection(connection)
+{
+    dbus_register_methods();
+}
+
+void RepoConf::dbus_register_methods()
+{
+    const std::string objectPath = "/org/rpm/dnf/v0/rpm/RepoConf";
+    const std::string interfaceName = "org.rpm.dnf.v0.rpm.RepoConf";
+
+    dbus_object = sdbus::createObject(connection, objectPath);
+    dbus_object->registerMethod(interfaceName, "list", "as", "aa{sv}", [this](sdbus::MethodCall call) -> void {this->list(call);});
```

Comment:
Methods start with an upper case by convention.

Also, I wonder if a method is the right way to expose this. At first sight, getting the list of repos should be a fairly cheap operation? Then this may be more useful to expose as a [property](https://dbus.freedesktop.org/doc/dbus-api-design.html#api-design). You could then watch it for changes, and emit `PropertiesChanged` when the repositories change in the system, so that clients can react to it. Of course you can also do this with a method -- whenever it changes, you could send out a  custom `RepositoriesChanged` signal, but this is an extra roundtrip, and clients need to write custom code for it as well. OTOH client libraries often automatically handle `PropertiesChanged` already.